### ISSUE TITLE
Deprecations and render prop performance

### ISSUE THREAD

**Issue by GeKorm** [thebuilder/react-intersection-observer#90]
Hi there. Thanks for creating this great library. 

The v6 release deprecates traditional render methods. Unfortunately, we have a use case where we can't use render props due to performance concerns.  
  
We have to render a list of items which is the triple whammy:
- It's big with complex items
- It re-renders frequently and has animations
- It is interactive

We've benchmarked it to death, including using render props for another HOC we have, and it's just not possible without adverse effects on performance. It's already slow on underpowered devices, every little bit counts. 

We use react-intersection-observer as a virtualized rendering solution because for this specific list it's much faster than react-virtualized (initialization time especially). Would you consider un-deprecating at least the "child as React.Node" method? 

Thanks

**Comment by thebuilder** [comment]
Hey George, thanks for the feedback!

Can you provide some insight into how you are using it? It sounds like there's room for improvement in how you use the render props. :) I haven't heard of them causing performance issues before.

Moving the inner rendering to a separate component/method could help a lot.

The `<Observer>` should only trigger a render when the inView changes - so once it enters, and once it leaves the viewport. This shouldn't be a performance issue.

I'm open to keep the React.Node support if it makes sense (and it still works in 6.0.0 with a warning), but i'd really like for developers to make a choice about the DOM structure.

**Comment by thebuilder** [comment]
See this blog entry by Kent C. Dodds: https://blog.kentcdodds.com/answers-to-common-questions-about-render-props-a9f84bb12d5d

> I should add one thing. If you’re really concerned about inlining your render prop function and perf implications of that, then don’t inline the function! :)

```jsx
class MyComp extends React.Component {
  renderInner = ({inView, ref}) => (
    <div ref={ref}>{/* Your UI stuff here */}</div>
  )
  render() {
    return (
      <Observer>
        {this.renderInner}
      </Observer>
    )
  }
}
```

**Comment by GeKorm** [comment]
Having the function as a method isn't completely free either, it still takes time to bind this method when initializing each the component. Moving it to an inner component wouldn't help either. Again, this is a list that updates multiple times a second, and can completely unmount/mount in one click.

The performance requirements are such that in the future we'll move to manually managing the list outside of React, unless React 17 improves things dramatically. But for now I really appreciate the React.Node support and I'm open to contribute to its maintenance.

**Comment by thebuilder** [comment]
See #91 - It's not a big thing to support it, but it's just important that developers are aware that it adds a wrapping `<div>`.

If you trying to replace `react-virtualized` with just rendering all elements using intersection observers, then you'll likely run it performance issues when scrolling.

**Comment by GeKorm** [comment]
Thanks! I'm debouncing and asynchronously applying the intersection-observer updates so it's no problem, it's much faster than react-virtualized 😄

**Comment by thebuilder** [comment]
Resolved with #91
